File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
count=0
word_list=['A', 'E', 'I', 'O', 'U']
    
def solution(word):
    DFS("",word)
    return answer

def DFS(word,target):
    global answer,count
    if word==target:
        answer=count
        return
    if len(word)>=5:
        return;
    for i in range (len(word_list)):
        count+=1
        DFS(word+word_list[i],target)

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    static char[] word_list={'A', 'E', 'I', 'O', 'U'};
    static int count=0,answer=0;
    public int solution(String word) {
        DFS("",word);
        return answer;
    }
    public void DFS(String word,String target){
        if (word.equals(target)){
                answer=count;
                return;
            }
        if (word.length()>=5){
            return;
        }
        for (int i=0;i<word_list.length;i++){
            count++;
            String w=word+word_list[i];
            DFS(w,target);
        }
    }
}